### ISSUE TITLE
Unable to flash firmware for Sonoff Dongle E

### ISSUE THREAD

**Issue by mckrosrw** [darkxst/silabs-firmware-builder#111]
Hi there,

I am very new to this. I bought a dongle recently and read that it is advised to update the firmware. I tried https://darkxst.github.io/silabs-firmware-builder/ but it could not detect the firmware. I tried https://github.com/NabuCasa/universal-silabs-flasher, but I get a lot of error messages as per below. Any idea how I can resolve this? I am on a Windows laptop and a novice in this area:

![ErrorMessages](https://github.com/user-attachments/assets/6e79df84-d84b-4158-bb46-4d0b42d7eed1)

If I can't get this to work, I guess I could try manually disassembling the dongle and using the reset bootloader (once I buy a screwdriver small enough for the screws :)). Otherwise, maybe the dongle just is not working.

thanks for any help

**Comment by darkxst** [comment]
This error probably indicates an issue with the installation of the flasher. Did you install it using `pip`?

**Comment by mckrosrw** [comment]
Thanks, yes I installed with pip. Should I remove the universal-silabs-flasher packlage and reinstall or could there be any other cause of the errors?

**Comment by darkxst** [comment]
You could try reinstall it, however I not very familiar with windows myself. 

If you are using Home Assistant you could also try flashing using the Silabs Flasher Add-on

**Comment by mckrosrw** [comment]
Yes, installed a more up-to-date version of python for windows (https://www.python.org/downloads/windows/), but after a  lot of time still getting the same error. Also, tried on a separate Windows machine without success. I did not know about the add-on, thanks for that. I will that a go before disassembling the sonoff dongle. Thanks again!

**Comment by Nosnahc** [comment]
> Yes, installed a more up-to-date version of python for windows (https://www.python.org/downloads/windows/), but after a lot of time still getting the same error. Also, tried on a separate Windows machine without success. I did not know about the add-on, thanks for that. I will that a go before disassembling the sonoff dongle. Thanks again!


Have you tried with [this windows driver](https://learn.adafruit.com/how-to-install-drivers-for-wch-usb-to-serial-chips-ch9102f-ch9102/windows-driver-installation)?

**Comment by darkxst** [comment]
> Have you tried with [this windows driver](https://learn.adafruit.com/how-to-install-drivers-for-wch-usb-to-serial-chips-ch9102f-ch9102/windows-driver-installation)?

That is the driver for the original Dongle-E. Now they are using Silicon Labs cp2102 chips.
